hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieFileWriterFactory.java

```diff
@@ -43,39 +40,18 @@
 
 public class HoodieFileWriterFactory {
 
-  private static HoodieFileWriterFactory getWriterFactory(HoodieRecord.HoodieRecordType recordType) {
```

Comment:
Let's create a JIRA ticket to make the `HoodieFileReaderFactory` and `HoodieFileWriterFactory` interface or abstract class so APIs to create new reader and writer should be abstract.

Reply:
Not exactly sure what you want

Reply:
`HoodieFileReaderFactory` and `HoodieFileWriterFactory` contain such methods that throw `UnsupportedOperationException`.  Instead, such methods should be abstract and the factory classes should also be made abstract or interface.
```
protected HoodieFileReader newParquetFileReader(StorageConfiguration<?> conf, StoragePath path) {
    throw new UnsupportedOperationException();
  }
```